### ISSUE TITLE
Improve error message of systemctl --user when XDG_RUNTIME_DIR is not set correctly due "su" behavior

### ISSUE THREAD

**Issue by lnykryn** [systemd/systemd#17901]
We have a lot of users that are confused by the following behaviour:

```
[lnykryn@X1 ~]$ systemctl --user is-active ddbd.socket
active
[lnykryn@X1 ~]$ ssh root@localhost
Last login: Tue Dec  8 14:55:55 2020 from ::1
[root@X1 ~]# su lnykryn
[lnykryn@X1 root]$ systemctl --user is-active ddbd.socket
Failed to connect to bus: Permission denied
```

The error message here is not saying what is actually wrong (su does not start pam modules that are setting env)
We should suggest that root either needs to use ssh or some other method of login that uses the full PAM stack or systemctl -M user@.host.  (in the second case we need to fix #14580 first )

**Comment by keszybz** [comment]
ab4a88eb920e2f64a79a60c1ea9aecb7907a9635 improved the situation. It'd be good to retest with v247.

**Comment by poettering** [comment]
BTW, if you use "su - lnykryn", then you get a different error. The commit @keszybz referenced only affects the "su -" case. The simple "su" case @lnykryn originally posted leaves $XDG_RUNTIME_DIR of root set, which then means that systemctl right-fully gets told to fuck off with a permission error, since unpriv users should not be able to connect to root's dbus instance.

I figure we should have some helper call that validates if $XDG_RUNTIME_DIR actually points to a dir owned by our own user identity. And if not, we should return some recognizable error, maybe -EXDEV or so, so thta we can then ultimately generate a better error message for it.

**Comment by poettering** [comment]
Fix waiting in #17967
